src/hooks/useRepoService.ts

```diff
@@ -0,0 +1,62 @@
+'use client';
+
+import { useAuth } from './useAuth';
+
+export function useRepoService() {
+  const { user } = useAuth();
+
+  async function trackRepoForUser(githubRepoId: string) {
+    if (!user) {
+      throw new Error('User not authenticated');
+    }
+
+    const request = await fetch(`/api/users/${user.id}/repos`, {
+      method: 'POST',
+      body: JSON.stringify({ githubRepoId }),
+      headers: {
+        'Content-Type': 'application/json',
+      },
+    });
+
+    const { success } = await request.json();
+    if (!success) {
+      throw new Error('Failed to track repo for user');
+    }
+  }
+
+  async function fetchTrackedRepos(): Promise<GitHubRepo[]> {
+    if (!user) {
+      throw new Error('User not authenticated');
+    }
+
+    const request = await fetch(`/api/users/${user.id}/repos`, {
+      method: 'GET',
+    });
+
+    const { data, success } = await request.json();
+
+    if (!success) {
+      throw new Error('Failed to fetch tracked repos');
+    }
+    return data;
+  }
+
+  async function untrackRepoForUser(githubRepoId: string) {
+    if (!user) {
+      throw new Error('User not authenticated');
+    }
+    const request = await fetch(`/api/users/${user.id}/repos/${githubRepoId}`, {
+      method: 'DELETE',
+    });
+
+    const { success } = await request.json();
+    if (!success) {
+      throw new Error('Failed to untrack repo for user');
+    }
+  }
+  return {
+    trackRepoForUser,
```

Comment:
So, instead of "track" and "untrack" it would be "add" or "create" and "remove" or "delete". The word "track" has implications beyond our use case, so sticking to more conventional naming choices is a good idea here.